providers/desec/protocol.go

```diff
@@ -16,9 +18,9 @@ const apiBase = "https://desec.io/api/v1"
 
 // Api layer for desec
 type desecProvider struct {
-	domainIndex      map[string]uint32 //stores the minimum ttl of each domain. (key = domain and value = ttl)
-	nameserversNames []string
-	creds            struct {
+	domainIndex            map[string]uint32 //stores the minimum ttl of each domain. (key = domain and value = ttl)
+	domainIndexInitialized bool              // if the domainIndex was fully initialized once
```

Comment:
Do we need `domainIndexInitalized`?  Could we simply [check if it is non-nil?](https://stackoverflow.com/questions/31339249/check-if-a-map-is-initialised-in-golang)

Reply:
I've changed it.